powerline.el

```diff
@@ -443,18 +448,13 @@ static char * %s[] = {
 
 ;;;###autoload (autoload 'powerline-narrow "powerline")
 (defpowerline powerline-narrow
-  (let (real-point-min real-point-max)
-    (save-excursion
-      (save-restriction
-        (widen)
-        (setq real-point-min (point-min) real-point-max (point-max))))
-    (when (or (/= real-point-min (point-min))
-              (/= real-point-max (point-max)))
-      (propertize "Narrow"
-                  'mouse-face 'mode-line-highlight
-                  'help-echo "mouse-1: Remove narrowing from the current buffer"
-                  'local-map (make-mode-line-mouse-map
-                              'mouse-1 'mode-line-widen)))))
+  (when ;; (buffer-narrowed-p) introduced in Emacs 24.3.
+      (/= (- (point-max) (point-min)) (buffer-size))
```

Comment:
so this doesn't move the cursor when you do point-min and point-max?

Reply:
Not in any Emacs I'm aware of - `point-min` and `point-max` are some of the lowest-level primitives for inspecting buffers. (And, that's the same implementation as `buffer-narrowed-p`, which doesn't save any excursion.)